src/creator-logic.coffee

```diff
@@ -117,33 +145,61 @@ Namespace('Wordguess').CreatorLogic = do ->
 		return questionsAnswers
 
 	buildSaveData = (titleValue) ->
-		if manuallyHide is on
-			manualSkippingIndices = []
 
-			editables = document.querySelectorAll('#editable span')
-			for edit, i in editables
-				if edit.classList.contains 'manually-selected'
-					manualSkippingIndices.push i
+		manualSkippingIndices = Array.from(hiddenWordsIndices)
 
-			wordsToSkip = -1
 
 		paragraph = cleanParagraph(document.getElementById('paragraph').value)
 		questionsAnswers = buildQuestionsAnswers(paragraph)
 
+		currentMode = Wordguess.CreatorEvents.getCurrentMode()
+
 		return qset =
 			'questions_answers'     : questionsAnswers
 			'title'                 : replaceTags(document.getElementById('title').value)
 			'paragraph'             : replaceTags(paragraph.join ' ')
 			'wordsToSkip'           : wordsToSkip
 			'manualSkippingIndices' : manualSkippingIndices
+			'mode'				    : currentMode
```

Comment:
unfortunately, I don't think I can get around not adding another field to the qset structure. The current implementation has a variable, wordsToSkip, that is set to -1 if the manual selection was chosen otherwise a number if automatic selection was selected. Therefore, there is no information in the qset that saves the value for wordsToSkip on the creator side if manual selection was chosen. Therefore, I created another field that will only be used on the creator side storing the value for previous automatic values and if it's not present due to a qset being used from a previous version, I provided a default value